docs/current_docs/configuration/custom-runner.mdx

```diff
@@ -76,3 +79,89 @@ and responses will be sent in plaintext over the wire from the Dagger CLI to
 the runner.
 :::
 
+## GPU support
+
+:::warning
+GPU support is currently experimental and only works with NVIDIA GPUs.
+:::
+
+In order to use a GPU, Dagger needs a custom, GPU-enabled runner. Assuming Dagger is already installed on a GPU-capable host, use the instructions below to replace the default runner with a GPU-enabled runner.
+
+:::note
+The sections below provide instructions for the local host and for cloud infrastructure providers [Fly.io](https://fly.io/) and [Lambda Labs](https://lambdalabs.com/). These instructions can be adapted for use on other cloud providers, so long as the host has access to an NVIDIA GPU.
+:::
+
+<Tabs>
+<TabItem value="Local host">
+
+Use the following commands to deploy a GPU-enabled Dagger runner on the local host:
+
+```shell
+VERSION=$(dagger version | cut -d' ' -f2)
+docker rm -f dagger-engine-${VERSION} 2>/dev/null && docker run --gpus all -d --privileged -e _EXPERIMENTAL_DAGGER_GPU_SUPPORT=true --name dagger-engine-${VERSION} registry.dagger.io/engine:${VERSION}-gpu -- --debug
```

Comment:
This failed for me locally:

```console
VERSION=$(dagger version | cut -d' ' -f2)
docker rm -f dagger-engine-${VERSION} 2>/dev/null && docker run --gpus all -d --privileged -e _EXPERIMENTAL_DAGGER_GPU_SUPPORT=true --name dagger-engine-${VERSION} registry.dagger.io/engine:${VERSION}-gpu -- --debug
dagger-engine-v0.15.0
Unable to find image 'registry.dagger.io/engine:v0.15.0-gpu' locally
v0.15.0-gpu: Pulling from engine
6414378b6477: Pull complete
f3f31896e111: Pull complete
dd53e36ba253: Pull complete
e40a4fc2c162: Pull complete
992090c3d7e8: Pull complete
05867476f9b2: Pull complete
49fbc24af143: Pull complete
b1b5e807b02c: Pull complete
cd5da30da36f: Pull complete
7e59763505e5: Pull complete
f81e1b632d07: Pull complete
e48577751f15: Pull complete
b2e287c09363: Pull complete
4bb703cb8595: Pull complete
755d3653d9d7: Pull complete
349ee5a316c0: Pull complete
71a8a39074a8: Pull complete
08565f4d9a3b: Pull complete
358d5c9d7fa6: Pull complete
ed00a6c26afb: Pull complete
0f1e06a9a82e: Pull complete
603956c2914f: Pull complete
5099601ea22f: Pull complete
fa825183aa6d: Pull complete
71da08f2e62a: Pull complete
e9f5fcd5d195: Pull complete
8dd8fbb42c01: Pull complete
8f7cbd4e25f3: Pull complete
01349efd6c30: Pull complete
1f88f3920a66: Pull complete
904ff5413ec9: Pull complete
ef329be288b5: Pull complete
976ca8844b9b: Pull complete
Digest: sha256:d8b8a190874a0e13f8b378fd56c58f2180016250f42917886decadf205fcfe2f
Status: Downloaded newer image for registry.dagger.io/engine:v0.15.0-gpu
20a796f92e54b53c11659993ddf600c27b5222e636248a9d2c3c9f2af7c89680
docker: Error response from daemon: could not select device driver "" with capabilities: [[gpu]].
```

I **think** that [NVIDIA Container Toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#installing-the-nvidia-container-toolkit) needs to be installed too. Trying this out now

Reply:
These are all the steps that I had to go through to make this command work:

```console
curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
  && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
    
sudo apt-get update

sudo apt-get install -y nvidia-container-toolkit

sudo nvidia-ctk runtime configure --runtime=docker

sudo systemctl restart docker
```

This command now worked for me:

```console
VERSION=$(dagger version | cut -d' ' -f2)
docker rm -f dagger-engine-${VERSION} 2>/dev/null && docker run --gpus all -d --privileged -e _EXPERIMENTAL_DAGGER_GPU_SUPPORT=true --name dagger-engine-${VERSION} registry.dagger.io/engine:${VERSION}-gpu -- --debug
25aa9fff0792437398a1fadde8b08d7d0fbd7bf6c06d805140756792f294d6ea
```

The only action item from me is to link to https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#installing-the-nvidia-container-toolkit in this context.

Reply:
This also worked: https://dagger.cloud/dagger/traces/7cb423439cefede8f8181c98ea631c97

```console
dagger -m github.com/samalba/dagger-modules/nvidia-gpu call has-gpu

✔ connect 0.2s
✔ load module 58.0s
✔ parsing command line arguments 0.0s

✔ gpu: Gpu! 0.0s
✔ .hasGpu: Boolean! 4.9s

true

Full trace at https://dagger.cloud/dagger/traces/7cb423439cefede8f8181c98ea631c97
```

And this as well: https://dagger.cloud/dagger/traces/b665a7469d2776dc5910075f91559636

```console
dagger -m github.com/samalba/dagger-modules/nvidia-gpu call ollama-run --prompt "What color is the sky?"
✔ connect 0.2s
✔ load module 2.3s
✔ parsing command line arguments 0.0s

✔ gpu: Gpu! 0.0s
✔ .ollamaRun(prompt: "What color is the sky?"): String! 1m53s

The color of the sky can vary depending on the time of day and atmospheric conditions. Here are some common colors of the sky:

1. **Clear blue sky**: On a sunny day, the sky typically appears as a bright blue, with shades ranging from light blue to deep blue.
2. **Sunrise and sunset**: During these times, the sky can take on hues of orange, pink, red, and purple, depending on the amount of atmospheric particles and dust in the air.
3. **Cloudy sky**: On an overcast day, the sky may appear gray or white, with clouds covering most of the sky.
4. **Twilight**: In the early morning and late evening hours, the sky can take on a range of colors, including pink, orange, purple, and blue.
5. **Atmospheric conditions**: Dust, pollution, and other atmospheric particles can scatter light in different ways, creating shades of red, orange, or yellow in the sky.

So, to summarize, the color of the sky is not always just one specific color – it can vary greatly depending on the time of day and atmospheric conditions!

Full trace at https://dagger.cloud/dagger/traces/b665a7469d2776dc5910075f91559636
```

Reply:
I can't deploy on fly, using my personal org: https://github.com/samalba/dagger-modules/blob/main/nvidia-gpu/main.go#L31 -- it is linked to dagger

Reply:
I resolved it in the module, per our discussion on discord.